project/shacl/modos_schema.shacl.ttl

```diff
@@ -126,291 +140,335 @@ modos:VariantSet a sh:NodeShape ;
             sh:order 5 ;
-            sh:path schema1:description ] ;
+            sh:path schema1:description ],
+        [ sh:datatype xsd:anyURI ;
+            sh:description "The URI from which a resource or dataset was obtained or derived." ;
+            sh:maxCount 1 ;
+            sh:nodeKind sh:Literal ;
+            sh:order 1 ;
+            sh:path modos:source_uri ] ;
     sh:targetClass modos:ReferenceSequence .
 
 modos:ReferenceGenome a sh:NodeShape ;
     sh:closed true ;
     sh:description "Reference assembly of a given genome, consisting of a collection of congiguous sequences (contigs)." ;
     sh:ignoredProperties ( rdf:type ) ;
-    sh:property [ sh:class modos:ReferenceSequence ;
-            sh:description "Denotes that a sequence belongs to a collection (e.g. a reference genome)." ;
-            sh:nodeKind sh:IRI ;
-            sh:order 1 ;
-            sh:path modos:has_sequence ],
+    sh:property [ sh:datatype xsd:string ;
+            sh:description "The path to access a resource, on a network or local filesystem. Can be a path relative to the root of the digital object, a URL, or an absolute path." ;
+            sh:maxCount 1 ;
+            sh:minCount 1 ;
+            sh:nodeKind sh:Literal ;
+            sh:order 0 ;
+            sh:path modos:data_path ],
         [ sh:datatype xsd:string ;
-            sh:description "A human-readable description for a thing" ;
+            sh:description "A human-readable name for a thing" ;
             sh:maxCount 1 ;
-            sh:order 7 ;
-            sh:path schema1:description ],
+            sh:nodeKind sh:Literal ;
+            sh:order 6 ;
+            sh:path schema1:name ],
         [ sh:datatype xsd:anyURI ;
             sh:description "The URI from which a resource or dataset was obtained or derived." ;
             sh:maxCount 1 ;
+            sh:nodeKind sh:Literal ;
             sh:order 3 ;
             sh:path modos:source_uri ],
+        [ sh:class modos:ReferenceSequence ;
+            sh:description "Denotes that a sequence belongs to a collection (e.g. a reference genome)." ;
+            sh:nodeKind sh:IRI ;
+            sh:order 1 ;
+            sh:path modos:has_sequence ],
         [ sh:datatype xsd:integer ;
             sh:description "The taxid number describing the taxonomic range of a sample." ;
+            sh:nodeKind sh:Literal ;
             sh:order 2 ;
             sh:path modos:taxon_id ],
         [ sh:datatype xsd:string ;
-            sh:description "The path to access a resource, on a network or local filesystem. Can be a path relative to the root of the digital object, a URL, or an absolute path." ;
-            sh:maxCount 1 ;
-            sh:minCount 1 ;
-            sh:order 0 ;
-            sh:path modos:data_path ],
-        [ sh:datatype xsd:string ;
-            sh:description "A human-readable name for a thing" ;
+            sh:description "A string specifying the release or version of a software or resource." ;
```

Comment:
sounds like a regex could be used here to check for #.#.# pattern.

Reply:
yeah unfortunately, not all software follow semver conventions :p